clang/lib/Driver/ToolChains/CommonArgs.cpp

```diff
@@ -490,6 +490,16 @@ void tools::AddLinkerInputs(const ToolChain &TC, const InputInfoList &Inputs,
     else
       A.renderAsInput(Args, CmdArgs);
   }
+  if (const Arg *A = Args.getLastArg(options::OPT_fveclib)) {
+    if (A->getNumValues() == 1) {
+      StringRef V = A->getValue();
+      if (V == "ArmPL") {
+        CmdArgs.push_back(Args.MakeArgString("-lamath"));
+        CmdArgs.push_back(Args.MakeArgString("-lm"));
```

Comment:
do these flags and library names work on Windows?

Reply:
I don't _think_ that -fveclib= works on Windows in the first place. I will check though.
There should probably be an `if windows ...` check around these too regardless, in case we do enable it on future in Windows.

Reply:
I suppose the actual answer to your question is no, as Windows doesn't have a separate libm, and the logic for linking to amath will probably be more complicated because you need a version of it matching your libc (the debug/release/static/dynamic grid that Windows makes you deal with)

Reply:
Personally I think things like needing to know the dependencies required by a third party library show why we shouldn't be baking third party libraries into the driver in the first place.  How the library is built is not under llvm's control nor are the libraries part of a recognised operating system.  Are there any existing examples where other third party libraries are baked in?

Reply:
We actually do support this on Windows, I was wrong above. So we should not be breaking this for Windows users.

I suppose the other option is to say that if someone passes `-fveclib=ArmPL` they will also have to pass the right flags to link to ArmPL and libamath themselves, rather than doing anything in the driver automatically? Assuming that's what @paulwalker-arm means, I'd probably agree and say that should be on the user to sort out and not on the driver.

Reply:
> do these flags and library names work on Windows?

I've reduced it to Linux only.

Reply:
> Flang and ArmPL both support Windows. I don't see why Windows should get second-class support. I would prefer to see this PR extended to also include any flags required on Windows.

This code is common for flang and clang.

@DavidTruby as an Windows expert, what flags would you recommend for Windows here?

Reply:
This function you're changing here (AddLinkerInputs) isn't actually called on Windows so no need to actually add an exclusion there. ArmPL is also supported on MacOS, but there it should use the same flags as it does on Linux. So I think what you had before is fine.

You'll need something similar for supporting both clang and flang on Windows, probably somewhere around where the other Windows on Arm options are: https://github.com/llvm/llvm-project/blob/927032807dfdca5d94eb0a8707d38b605e95e407/clang/lib/Driver/ToolChains/MSVC.cpp#L86

Based on a quick glance at the ArmPL docs for the library name, the flag you'll need here is:
"--dependent-lib=amath"

Reply:
> Flang already has a hack for a different veclib implementation on darwin

The difference with the one instance of doing this is that Accelerate is a core part of that environment, much like libm is a core part of the linux environment.  libamath is not core to any environment.  I recommend seeking advice from the wider clang community before landing this work.

Reply:
> This function you're changing here (AddLinkerInputs) isn't actually called on Windows

So much about putting 'Common' in the filename (CommonArgs.cpp)

Reply:
MinGW here doesn't really mean MinGW (I realise this is quite confusing). It more specifically  means "using glibc, ld.bfd and libstdc++". This isn't supported on AArch64 at all, and is mostly being phased out on X86 too I believe. Only the "new style" mingw-w64 is supported on AArch64 (using libucrt and link.exe).

There's no Cygwin on AArch64 either.
I suppose more specifically I should have said that this function will never be called on Windows _on Arm_ 🙂